### ISSUE TITLE
Warn against user defined `__magic__` methods

### ISSUE THREAD

**Issue by jpy-git** [PyCQA/pylint#5940]
### Current problem

I'm currently working on a PR (#5938) that warns users against manually invoking magic methods.

This led me to wonder what happens if a user defines their own method using the `__dunder__` naming convention.

e.g.
```python
class A:
    def __foo__():
        pass
```

According to the [docs](https://docs.python.org/3/reference/lexical_analysis.html#reserved-classes-of-identifiers)
>`__*__`
System-defined names, informally known as “dunder” names. These names are defined by the interpreter and its implementation (including the standard library). Current system names are discussed in the [Special method names](https://docs.python.org/3/reference/datamodel.html#specialnames) section and elsewhere. More will likely be defined in future versions of Python. Any use of `__*__` names, in any context, that does not follow explicitly documented use, is subject to breakage without warning.

So we should warn users against ever creating a method with that naming convention!

### Desired solution

New lint checker that checks for any method definition with a `__dunder__` style name that is not an official python `__dunder__` method.
